### PR TITLE
Create additional validation checks for use in user creation

### LINKED ISSUE #74: Validations for input createUser
## Description

Implement input validations for the 'createUser' function, encompassing checks for both the UID, username and email.

## Requirements

- [x] Research validations, what kind of validations are best practice working with RESTful API.
- [x] Implemented additional validations to check for empty inputs, inputs of a certain length, and checking of uid, username and emails according to common norms.

### PR DESCRIPTION
**Pull request by oddpharmacy**
# Description

Implement additional checks which cover common scenarios to improve the overall security of the user creation process. This includes length and empty character checks for the uid, username and email user inputs.

# Closes issue(s)

Resolves #74 

# How to test?

### Test using Postman

- Start the server in your local environment. -> `npm run dev`
- Ensure that you have run the necessary prisma scripts, created a local database, and have linked the database url in your env file.
- Open Postman and make a POST req to the _signup_ route. -> `http://localhost:8080/signup`

**_Case 1: A successful request and creation of a new user_**
- Paste the following content as the body to your POST request:
`{
  "username": "totoroo",
  "email": "totoroo@example.com",
  "uid": "aBcDeF1234567890aBcDeF1234--"
}
`
- You should receive an 'id' and 'username' back as a response.

_**Case 2: Invalid inputs entered / Duplicate users**_
- Paste the following content as the body to your POST request:
`{
  "username": "totoroo",
  "email": "totoroo@example.com",
  "uid": "a"
}
`
- You should receive the following error as a response: `Error: Something went wrong. Please check your inputs.`
- Different error messages will pop up depending on the user's inputs. 

# Errors
If you encounter a Typescript error when trying to run the server, make the fix below in _login.service.ts_ on line 14:
`    return loginModel.updateAccessTokenLogin(userId);
`

# Changes include

1. isNotEmptyInput, isValidUID, isValidEmail, and isValidUsername functions.

# Checklist

- [x] I have tested this code

### CODE REVIEW

File: src/utils/validation.ts
Comment (shonancanuck) [inline src/utils/validation.ts:12]: Depending on the particular pattern for UID we're using, this could potentially be a little more stringent, but I'm not even sure that's necessary.

**Review by shonancanuck — COMMENTED** [review]
Everything looks great to me. I made one comment, but it's more an observation than a suggestion.

File: src/service/signup.service.ts
Comment (Anissa3005) [inline src/service/signup.service.ts:39]: I think there is a better to handle the errors but this should be done in an other issue.  It's okay for now but maybe create a To Do in your PR comment and add it there

**Review by Anissa3005 — CHANGES_REQUESTED** [review]
Great Job!

when I run Eslint i saw some errors can you please fixed them and commit them before merging. Some errors are from an other PR but Chai and Me fixed them so maybe wait on his PR to be merged so you can rebase.

**Comment by oddpharmacy** [comment]
Commits messed up. New branch and subsequent PR to replace.

**Pull request closed without merging**
